application/client/src/app/ui/views/toolbar/chart/zoomer/cursor/state.ts

```diff
@@ -0,0 +1,280 @@
+import { IlcInterface } from '@service/ilc';
+import { ViewContainerRef } from '@angular/core';
+import { getChartService, Service, IPosition } from '../../service';
+import { ChangesDetector } from '@ui/env/extentions/changes';
+import { Session } from '@service/session';
+import { stop } from '@ui/env/dom';
+
+enum EChangeKind {
+    move = 'move',
+    left = 'left',
+    right = 'right',
+    none = 'none',
+}
+
+const MIN_SIZE_WIDTH = 20;
+
+export class State {
+    public _ng_width: number = -1;
+    public _ng_left: number = 0;
+
+    private _parent!: IlcInterface & ChangesDetector;
+    private _session!: Session;
+    private _service!: Service;
+    private _leftOffset = () => 0;
+    private _vcRef!: ViewContainerRef;
```

Comment:
you can store is as `HTMLElement` no need to store reference to Angular container